lib/src/main/java/org/exoplatform/social/addons/updater/RDBMSMigrationManager.java

```diff
@@ -140,6 +156,10 @@ public void run() {
                       && !MigrationContext.isIdentityDone()) {
                 getIdentityMigrationService().start();
                 updateSettingValue(MigrationContext.SOC_RDBMS_IDENTITY_MIGRATION_KEY, MigrationContext.isIdentityDone());
+
+                if (useMigrationIdentityStorage && MigrationContext.isIdentityDone()) {
+                  identityManager.setIdentityStorage(identityStorage);
```

Comment:
> If something fails and it stops the identities migration, the implementation MigrationIdentityStorage will be used and when users will need non-migrated identities, they will be migrated on the fly. And when PLF is restarted, original implementation will be used by default, then it will be replaced by the MigrationIdentityStorage implementation again since the migration was not finished. Right ?

Yes, That's right

> Since identities can be migrated on the fly, we could end up in a situation where all the identities will be migrated but the MigrationContext is not aware of that. So the migration will be launched again at the next restart for nothing (since it will not migrate anything). Right ?
 
Yes, that's right too. After this, the MigrationContext will be marked as completed to avoid this loop next time.